### PR TITLE
Dev select between two wine rules

### PR DESCRIPTION
**Pull request by john-baxter**

### CODE REVIEW

File: wine.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+def score_wine_Collector_method(hand, standard_tokens_for_trail):
+  """Uses the Collector rule to calculate the number of points for wine tokens
+  
+  Each wine is worth 1 point for each different type of food you have at least one of
+
+  Could be called any time but to reflect the real game there is no running 
+  total of points; just a single calculation and comparison after the final ant 
+  has reached the anthill.
```
Comment (anaandresarroyo) [inline wine.py]: This sentence is correct, but not relevant to this function. This function doesn't really care when it is being called.

File: wine.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
+def score_wine_Collector_method(hand, standard_tokens_for_trail):
+  """Uses the Collector rule to calculate the number of points for wine tokens
+  
+  Each wine is worth 1 point for each different type of food you have at least one of
```
Comment (anaandresarroyo) [inline wine.py]:
```suggestion
  Each wine is worth 1 point for each different type of food you have at least one of.
```

File: wine.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  return wine_score
+
+def score_wine_Oenophile_method(hand, standard_tokens_for_trail):
+  """Uses the Oenophile rule to calculate the number of points for wine tokens
+
+  Each wine is worth one point for each wineyou have.
+
+  Could be called any time but to reflect the real game there is no running 
+  total of points; just a single calculation and comparison after the final ant 
+  has reached the anthill.
```
Comment (anaandresarroyo) [inline wine.py]: This sentence is correct, but not relevant to this function. This function doesn't really care when it is being called.

File: wine.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  """
+  wine_score = hand["wine"] *\
+    (len([food for food in hand\
+    if food in standard_tokens_for_trail and hand[food] > 0]))
+  return wine_score
+
+def score_wine_Oenophile_method(hand, standard_tokens_for_trail):
+  """Uses the Oenophile rule to calculate the number of points for wine tokens
+
+  Each wine is worth one point for each wineyou have.
```
Comment (anaandresarroyo) [inline wine.py]:
```suggestion
  Each wine is worth one point for each wine you have.
```

File: wine.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    which calles this also calls others where this is required. It is included here for concistency.
+    Keys are food types as strings
+    Values are integers.
+
+  Returns
+  -------
+  wine_score : (integer)
+    An integer showing the player's total points from wine. 
+  """
+  wine_score = hand["wine"] * hand["wine"]
```
Comment (anaandresarroyo) [inline wine.py:47]: Optional change: Powers are done with `**`.
```suggestion
  wine_score = hand["wine"] ** 2
```

**Review by anaandresarroyo — APPROVED** [review]
Looks good. Not many comments but I did look at all the files.

File: wine.py (hunk 5 above)
Comment (john-baxter) [reply wine.py:47]: I did check up on this first, and apparently n*n vs n**2 is not much different; but with smaller numbers n*n is slightly faster.

**Pull request merged**